commercetools-sdk-base/src/main/java/io/sphere/sdk/client/CorrelationIdGenerator.java

```diff
@@ -0,0 +1,31 @@
+package io.sphere.sdk.client;
+
+/**
+ * Generates correlation ids for each request.
+ *
+ * The format of the correlation id is defined as:
+ * {@code <projectKey> / <UUID> (/ customId)*}
+ */
+public interface CorrelationIdGenerator {
```

Comment:
concerning the documentation it has too much implementation detail. I would just expect from the interface to say it will generate a unique correlationId and has only the createCorrelationId method.